app/dash/Chains/Chain/ChainNew/index.js

```diff
@@ -131,8 +131,17 @@ export default ({
           enabled={chainValidation.valid}
           onClick={() => {
             if (chainValidation.valid) {
+              const nav = store('windows.dash.nav')
+              const chainsView = { view: 'chains', data: {} }
               link.send('tray:addChain', updatedChain)
-              link.send('tray:action', 'backDash')
+
+              // if previous navItem is the chains panel, go back
+              if (JSON.stringify(nav[1]) === JSON.stringify(chainsView)) {
```

Comment:
I think we can just check that the previous view was the chains view here, I don't think the data is relevant for this comparison